TLM/TLM/Custom/AI/CustomHumanAI.cs

```diff
@@ -1,374 +1,555 @@
-				}
+                Log._DebugIf(
+                    logTimedLights,
+                    () => $"CustomHumanAI.CustomCheckTrafficLights({nodeId}, {segmentId}): " +
+                    $"Custom simulation! pedestrianLightState={pedestrianLightState}, " +
+                    $"lights.InvalidPedestrianLight={lights.InvalidPedestrianLight}");
 #endif
-			}
-			// NON-STOCK CODE END //
-
-			switch (pedestrianLightState) {
-				case RoadBaseAI.TrafficLightState.RedToGreen:
-					if (stepWaitTime < 60u) {
-						return false;
-					}
-					break;
-				case RoadBaseAI.TrafficLightState.Red:
-				case RoadBaseAI.TrafficLightState.GreenToRed:
-					return false;
-			}
-			return true;
-		}
-
-		[RedirectReverse]
-		[MethodImpl(MethodImplOptions.NoInlining)]
-		private void PathfindFailure(ushort instanceID, ref CitizenInstance data) {
-			Log.Error($"HumanAI.PathfindFailure is not overriden!");
-		}
-
-		[RedirectReverse]
-		[MethodImpl(MethodImplOptions.NoInlining)]
-		private void PathfindSuccess(ushort instanceID, ref CitizenInstance data) {
-			Log.Error($"HumanAI.PathfindSuccess is not overriden!");
-		}
-
-		[RedirectReverse]
-		[MethodImpl(MethodImplOptions.NoInlining)]
-		private void Spawn(ushort instanceID, ref CitizenInstance data) {
-			Log.Error($"HumanAI.Spawn is not overriden!");
-		}
-
-		[RedirectReverse]
-		[MethodImpl(MethodImplOptions.NoInlining)]
-		private void GetBuildingTargetPosition(ushort instanceID, ref CitizenInstance data, float minSqrDistance) {
-			Log.Error($"HumanAI.GetBuildingTargetPosition is not overriden!");
-		}
-
-		[RedirectReverse]
-		[MethodImpl(MethodImplOptions.NoInlining)]
-		private void WaitTouristVehicle(ushort instanceID, ref CitizenInstance data, ushort targetBuildingId) {
-			Log.Error($"HumanAI.InvokeWaitTouristVehicle is not overriden!");
-		}
-
-		[RedirectReverse]
-		[MethodImpl(MethodImplOptions.NoInlining)]
-		private void ArriveAtDestination(ushort instanceID, ref CitizenInstance citizenData, bool success) {
-			Log.Error($"HumanAI.ArriveAtDestination is not overriden!");
-		}
-	}
+            }
+
+            // NON-STOCK CODE END
+            switch (pedestrianLightState) {
+                case RoadBaseAI.TrafficLightState.RedToGreen:
+                    if (stepWaitTime < 60u) {
+                        return false;
+                    }
+
+                    break;
+                case RoadBaseAI.TrafficLightState.Red:
+                case RoadBaseAI.TrafficLightState.GreenToRed:
+                    return false;
+            }
+
+            return true;
+        }
+
+        [RedirectReverse]
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        [UsedImplicitly]
+        private void PathfindFailure(ushort instanceId, ref CitizenInstance data) {
+            Log._DebugOnlyError("HumanAI.PathfindFailure is not overriden!");
```

Comment:
In this case it will create a false error in Release and this is not nice. There should be some better way to keep this method, like for example referring to it from somewhere. Maybe call some no-op function.

Reply:
I just double checked the Release build in decompiler. It is replaced with simple `ret` but not eliminated.
![image](https://user-images.githubusercontent.com/288724/61580361-78251280-ab11-11e9-887b-ae05c2854811.png)